components/widget/ratings/widget_ratings_list.lua

```diff
@@ -0,0 +1,129 @@
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Ratings/List
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Flags = require('Module:Flags')
+local Lua = require('Module:Lua')
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local OpponentDisplay = OpponentLibraries.OpponentDisplay
+
+local Widget = Lua.import('Module:Widget')
+local WidgetUtil = Lua.import('Module:Widget/Util')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local PlacementChange = Lua.import('Module:Widget/Standings/PlacementChange')
+local RatingsStorageFactory = Lua.import('Module:Ratings/Storage/Factory')
+
+---@class RatingsList: Widget
+---@field _base Widget
+---@operator call(table): RatingsList
+local RatingsList = Class.new(Widget)
+
+---@return Widget
+function RatingsList:render()
+	-- Simple check to verify that the input is a osdate
+	assert(self.props.date and self.props.date.wday, 'Invalid date provided')
+	---@type osdate
+	local date = self.props.date
+	local dateAsString = os.date('%F', os.time(date)) --[[@as string]]
+
+	local teamLimit = tonumber(self.props.teamLimit) or self.defaultProps.teamLimit
+	local progressionLimit = tonumber(self.props.progressionLimit) or self.defaultProps.progressionLimit
+
+	local getRankings = RatingsStorageFactory.createGetRankings {
+		storageType = self.props.storageType,
+		date = dateAsString,
+		id = self.props.id,
+	}
+	local teams = getRankings(teamLimit, progressionLimit)
+
+	local teamRows = Array.map(teams, function(team, rank)
+		local streakText = team.streak > 1 and team.streak .. 'W' or (team.streak < -1 and (-team.streak) .. 'L') or '-'
+		local streakClass = (team.streak > 1 and 'group-table-rank-change-up')
+			or (team.streak < -1 and 'group-table-rank-change-down')
+			or nil
+
+		local changeText = (not team.change and 'NEW') or PlacementChange { change = team.change }
+
+		local teamRow = WidgetUtil.collect(
+			HtmlWidgets.Td { attributes = { ['data-ranking-table-cell'] = 'rank' }, children = rank },
+			HtmlWidgets.Td { attributes = { ['data-ranking-table-cell'] = 'change' }, children = changeText },
+			HtmlWidgets.Td {
+				attributes = { ['data-ranking-table-cell'] = 'team' },
+				children = OpponentDisplay.BlockOpponent { opponent = team.opponent, teamStyle = 'hybrid' }
+			},
+			HtmlWidgets.Td { attributes = { ['data-ranking-table-cell'] = 'rating' }, children = team.rating },
+			HtmlWidgets.Td {
+				attributes = { ['data-ranking-table-cell'] = 'region' },
+				children = Flags.Icon(team.region) .. Flags.CountryName(team.region)
+			},
+			HtmlWidgets.Td {
+				attributes = { ['data-ranking-table-cell'] = 'streak' },
+				children = streakText,
+				classes = { streakClass }
+			}
+		)
+
+		local isEven = rank % 2 == 0
+		local rowClasses = { 'ranking-table__row' }
+		if isEven then
+			table.insert(rowClasses, 'ranking-table__row--even')
+		end
```

Comment:
from lookingt at the less changes shouldn't it be enough to replace instead of add?
```suggestion
		local rowClasses = { 'ranking-table__row' }
		if isEven then
			rowClasses = { 'ranking-table__row--even' }
		end
```

Reply:
Should be needed for eg td children to work